src/Controller/LoginController.php

```diff
@@ -35,9 +35,7 @@ public function loginAction() {
         $userName = $_POST['userName'];
         $password = $_POST['password'];
 
-        $user = new User();
-        
-        if (!$this->getUser($userName, $user)) {
+        if (!$user = $this->getUser($userName)) {
```

Comment:
Jei neklystu tokia `if` sąlyga bereikalingai naudoja atmintį, kadangi su `!` neigi ne `$user` objektą, bet priskyrimo rezultatą, kuris "po kapotu" yra išsaugomas į atskirą vietą atmintyje (lyg toks nematomas/paslėptas kintamasis su ta pačia reikšme kaip ir `$user`). 
Ir php bendruomenėje bendrai stengiamasi reikšmės priskyrimo statementų nenaudoti if sąlygose. Didele dalim turbūt dėl skaitomumo, nes dbr lengva pagalvoti kad darai `if(!$user)` nors iš tikrųjų nėra visai taip.  Kad nekalbėčiau sausai išleidžiu 6sekundes ir randu: https://stackoverflow.com/questions/317259/why-are-assignments-in-conditions-bad